src/components/DateTimePicker/DateTimePicker.js

```diff
@@ -76,81 +68,117 @@ const DateTimePicker = (props) => {
       onKeyDown();
     }
   };
+
   const handleTimeChange = (newTime) => {
     const dateWithOldTime = date ? new Date(date) : new Date();
     const hours = newTime ? newTime.getHours() : 0;
     const minutes = newTime ? newTime.getMinutes() : 0;
     const dateWithUpdatedTime = new Date(
-      dateWithOldTime.setHours(hours, minutes),
+      dateWithOldTime.setHours(hours, minutes, 0),
     );
     setDate(dateWithUpdatedTime);
   };
 
+  const clearTime = () => handleTimeChange(new Date(new Date().setHours(0, 0, 0)));
+
   const handleDateChange = (newDate) => {
-    setDate(newDate);
+    if (!newDate) {
+      return setDate(null);
+    }
+    const oldDate = date ? new Date(date) : new Date();
```

Comment:
Shouldn't this be? 

`const oldDate = date || new Date();`

In any case we are not going to modify oldDate, so this should probably work too.

Reply:
okay, after previous comment about mutability i just got into habit of making a new object every time, so that i dont forget. changed this.